SAODB-134 - MA - Adds skeleton test and framework updates for certificate submission

Michael-A-NC opened the pull request:
### Added
* Created stub for PUT certificate endpoint.
* Added skeleton test for this endpoint.
* Refactoring

Michael-A-NC committed:
Adds skelton test and framework updates for certificate submission

aca2492 commented on src/test/scala/uk/gov/hmrc/stubs/models/Company.scala:
```diff
@@ -0,0 +1,37 @@
+  uniqueTaxpayerReference: Option[String],
+  companyType: String,
+  financialYearEnd: Instant,
+  accountingPeriod: AccountingPeriod,
```
accountingPeriod can be removed? following understanding of company data in templates and within notification similarities

Michael-A-NC replied:
Thanks, good spot!

aca2492 reviewed (commented):
Looks good, only potential change is the accountingPeriod field in company following our discussion.

Michael-A-NC committed:
Adds faker for randomising data; refactoring

# Conflicts:
#	src/test/scala/uk/gov/hmrc/stubs/TestDataFactory.scala
#	src/test/scala/uk/gov/hmrc/support/ApiTestSupport.scala

Michael-A-NC committed:
Removes accountingPeriod from Company object as per review comment

Michael-A-NC committed:
Rebase and refactoring

aca2492 reviewed (approved):
All changes applied ready for merge.
